frontend/appflowy_flutter/lib/plugins/database/application/cell/cell_cache.dart

```diff
@@ -1,77 +1,51 @@
-part of 'cell_service.dart';
+import 'package:appflowy/plugins/database/application/row/row_service.dart';
+import 'package:appflowy_backend/log.dart';
 
-typedef CellContextByFieldId = LinkedHashMap<String, DatabaseCellContext>;
+import 'cell_controller.dart';
 
-class DatabaseCell {
-  dynamic object;
-  DatabaseCell({
-    required this.object,
-  });
-}
-
-/// Use to index the cell in the grid.
-/// We use [fieldId + rowId] to identify the cell.
-class CellCacheKey {
-  final String fieldId;
-  final RowId rowId;
-  CellCacheKey({
-    required this.fieldId,
-    required this.rowId,
-  });
-}
-
-/// GridCellCache is used to cache cell data of each block.
-/// We use GridCellCacheKey to index the cell in the cache.
+/// CellMemCache is used to cache cell data of each block.
+/// We use CellContext to index the cell in the cache.
 /// Read https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/architecture/frontend/grid
 /// for more information
 class CellMemCache {
-  final String viewId;
-
-  /// fieldId: {cacheKey: GridCell}
+  /// fieldId: {rowId: cellData}
   final Map<String, Map<RowId, dynamic>> _cellByFieldId = {};
-  CellMemCache({
-    required this.viewId,
-  });
+
+  CellMemCache();
 
   void removeCellWithFieldId(String fieldId) {
     _cellByFieldId.remove(fieldId);
   }
 
-  void remove(CellCacheKey key) {
-    final map = _cellByFieldId[key.fieldId];
-    if (map != null) {
-      map.remove(key.rowId);
+  void remove(CellContext context) {
+    if (_cellByFieldId.containsKey(context.fieldId)) {
+      _cellByFieldId[context.fieldId]!.remove(context.rowId);
     }
   }
 
-  void insert<T extends DatabaseCell>(CellCacheKey key, T value) {
-    var map = _cellByFieldId[key.fieldId];
-    if (map == null) {
-      _cellByFieldId[key.fieldId] = {};
-      map = _cellByFieldId[key.fieldId];
-    }
-
-    map![key.rowId] = value.object;
+  void insert<T>(CellContext context, T data) {
+    _cellByFieldId.putIfAbsent(context.fieldId, () => {});
+    _cellByFieldId[context.fieldId]![context.rowId] = data;
   }
 
-  T? get<T>(CellCacheKey key) {
-    final map = _cellByFieldId[key.fieldId];
-    if (map == null) {
+  T? get<T>(CellContext context) {
+    if (!_cellByFieldId.containsKey(context.fieldId)) {
```

Comment:
can these lines (32 to 48) refactor into this
```dart
    final value = _cellByFieldId[context.fieldId]?[context.rowId];
    if (value == null) {
      return null;
    }
```

Reply:
Yes. We'll lose the Log.error below, but I think it's not often that we get this error